### PR TITLE
Support tests on JDK > 17

### PR DESCRIPTION
**Pull request by parttimenerd**
### Description
Change the hard-coded use of `-source 7 -target 7` in the Makefile to using the Java version passed via `TARGET_RELEASE` (added for this very purpose in #655).

### Motivation and context
The compilation of some Java files still uses the hard-coded source and target version 7 in the Makefile, leading to issues with Java compilers from JDK 21 and newer, that don't support version 7 any more.

### How has this been tested?

By calling `make test TARGET_RELEASE=21` on JDK 21, it suceeds with this change without Java warnings.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -164,7 +164,7 @@ build/$(CONVERTER_JAR): $(CONVERTER_SOURCES) $(RESOURCES)
 	$(RM) -r build/converter
 
 %.class: %.java
-	$(JAVAC) -source 7 -target 7 -Xlint:-options -g:none $^
+	$(JAVAC) -source ${TARGET_RELEASE} -target ${TARGET_RELEASE} -Xlint:-options -g:none $^
```
Comment (krk) [inline Makefile:167]: Is `TARGET_RELEASE` different than `JAVA_TARGET`?

File: test/one/profiler/test/TestProcess.java [hunk 2]
```diff
@@ -85,6 +85,13 @@ public String[] inputs() {
         return this.inputs;
     }
 
+    private static boolean needsDynamicAgentLoadingFlag() {
+        String version = System.getProperty("java.version");
+        String[] versionParts = version.split("\\.");
+        int majorVersion = Integer.parseInt(versionParts[0]);
+        return majorVersion >= 21;
+    }
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:93]: Target JDK is not necessary the same as the host JDK.

File: test/one/profiler/test/TestProcess.java [hunk 3]
```diff
@@ -96,7 +103,10 @@ private List<String> buildCommandLine(Test test, Os currentOs) {
             cmd.add(substituteFiles(String.join(";", sh)));
         } else {
             cmd.add(System.getProperty("java.home") + "/bin/java");
-            cmd.add("-cp");
+            if (needsDynamicAgentLoadingFlag()) {
+                cmd.add("-XX:+EnableDynamicAgentLoading");
+            }
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:108]: Why is this option needed? `EnableDynamicAgentLoading` is true even in JDK 24.
For the cases when dynamic agent loading is disabled, I'll implement a different solution in jattach.

File: src/helper/one/profiler/Server.class
Comment (apangin) [inline src/helper/one/profiler/Server.class:1]: What has changed in binary files in this PR?

File: Makefile (hunk 1 above)
Comment (apangin) [inline Makefile:167]: I don't see where TARGET_RELEASE is defined.

File: test/one/profiler/test/TestProcess.java (hunk 3 above)
Comment (parttimenerd) [reply test/one/profiler/test/TestProcess.java:108]: Ok, I'll remove it

File: src/helper/one/profiler/Server.class
Comment (parttimenerd) [reply src/helper/one/profiler/Server.class:1]: I remove the change

File: Makefile (hunk 1 above)
Comment (parttimenerd) [reply Makefile:167]: good catch

File: test/one/profiler/test/TestProcess.java (hunk 2 above)
Comment (parttimenerd) [reply test/one/profiler/test/TestProcess.java:93]: oh, you're right

File: Makefile (hunk 1 above)
Comment (parttimenerd) [reply Makefile:167]: I replaced it `JAVA_TARGET`

**Comment by apangin** [comment]
I'm not sure what problem you are solving, to begin with.
I just cloned a fresh repo and successfully built it with JDK 23.
Helper classes are only recompiled if you change their sources, but if you do so, .class files (being part of the repo) should still be targeted the minimum supported version, which is currently JDK 7. I will probably change it to JDK 8 for the next major release.

**Comment by parttimenerd** [comment]
The problem is running the tests with JDK 21+.

**Comment by parttimenerd** [comment]
You're right, I must have made a different mistake.

**Pull request closed without merging**